File: Lib/Telerik/Examples/RssReader/ChangeCategoryForm.vb
Imports System.ComponentModel
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.UI

Namespace CS
	Partial Public Class ChangeCategoryForm
		Inherits Telerik.WinControls.UI.RadForm
		Public Sub New()
			InitializeComponent()
			radDropDownList1.DropDownStyle = RadDropDownStyle.DropDownList
		End Sub
		Public ReadOnly Property CheckBox() As RadCheckBox
			Get
				Return Me.radCheckBox1
			End Get
		End Property

		Public ReadOnly Property RadDropDownList() As RadDropDownList
			Get
				Return Me.radDropDownList1
			End Get
		End Property

		Public ReadOnly Property NewCategoryNameTextBox() As RadTextBox
			Get
				Return Me.radTextBox1
			End Get
		End Property

		Public ReadOnly Property NameTextBox() As RadTextBox
			Get
				Return Me.radTextBox2
			End Get
		End Property



		Private Sub radCheckBox1_ToggleStateChanged(ByVal sender As Object, ByVal args As Telerik.WinControls.UI.StateChangedEventArgs) Handles radCheckBox1.ToggleStateChanged
			If args.ToggleState = Telerik.WinControls.Enumerations.ToggleState.On Then
				radTextBox1.Enabled = True
				radDropDownList1.Enabled = False
			Else
				radTextBox1.Enabled = False
				radDropDownList1.Enabled = True
			End If
		End Sub

		Public Property ErrorText() As String

		Private Sub radButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles radButton1.Click
			If radCheckBox1.ToggleState = Telerik.WinControls.Enumerations.ToggleState.On AndAlso String.IsNullOrEmpty(radTextBox1.Text) Then
				errorProvider1.SetIconAlignment(radTextBox1, ErrorIconAlignment.MiddleLeft)
				errorProvider1.SetError(Me.radTextBox1, ErrorText)
				Return
			End If
			Me.DialogResult = DialogResult.OK
		End Sub

		Private Sub radButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles radButton2.Click
			Me.DialogResult = DialogResult.Cancel
		End Sub

		Protected Overrides Sub OnShown(ByVal e As EventArgs)
			MyBase.OnShown(e)
			Me.NameTextBox.TextBoxElement.TextBoxItem.HostedControl.Focus()
		End Sub


	End Class
End Namespace

File: Lib/Telerik/Examples/RssReader/ChangeCategoryForm.cs
using System;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.Enumerations;
using Telerik.WinControls.UI;

namespace RssReader
{
    public partial class ChangeCategoryForm : RadForm
    {
        public ChangeCategoryForm()
        {
            InitializeComponent();
            radDropDownList1.DropDownStyle = RadDropDownStyle.DropDownList;
        }

        public RadCheckBox CheckBox
        {
            get
            {
                return this.radCheckBox1;
            }
        }

        public RadDropDownList RadDropDownList
        {
            get
            {
                return this.radDropDownList1;
            }
        }

        public RadTextBox NewCategoryNameTextBox
        {
            get
            {
                return this.radTextBox1;
            }
        }

        public RadTextBox NameTextBox
        {
            get
            {
                return this.radTextBox2;
            }
        }
        
        private void radCheckBox1_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            if (args.ToggleState == ToggleState.On)
            {
                radTextBox1.Enabled = true;
                radDropDownList1.Enabled = false;
            }
            else
            {
                radTextBox1.Enabled = false;
                radDropDownList1.Enabled = true;
            }
        }

        public string ErrorText { get; set; }

        private void radButton1_Click(object sender, EventArgs e)
        {
            if (radCheckBox1.ToggleState == ToggleState.On && string.IsNullOrEmpty(this.radTextBox1.Text))
            {
                errorProvider1.SetIconAlignment(this.radTextBox1, ErrorIconAlignment.MiddleLeft);
                errorProvider1.SetError(this.radTextBox1, this.ErrorText);
                return;
            }
            this.DialogResult = DialogResult.OK;
        }

        private void radButton2_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
        }

        protected override void OnShown(EventArgs e)
        {
            base.OnShown(e);
            this.NameTextBox.TextBoxElement.TextBoxItem.HostedControl.Focus();
        }       
    }
}
